css-scroll-snap-2/Overview.bs

```diff
@@ -275,119 +275,244 @@ Snap Events {#snap-events}
+	associated with at most one <dfn>snapchangingTargetBlock</dfn> and at most one
+	<dfn>snapchangingTargetInline</dfn> in the block and inline axes respectively.
+
+	When asked to <dfn export for=Document>update snapchanging targets</dfn>
+	for a <a spec=css-scroll-snap lt="scroll snap container">snap container</a>,
+	|snapcontainer|, run these steps:
+
+	1. Let <var>doc</var> be |snapcontainer|'s associated {{Document}}.
+	1. Let <var>blockTarget</var> be the <a>snapchangingTargetBlock</a> that is
+		associated with |snapcontainer|.
+	1. Let <var>inlineTarget</var> be the <a>snapchangingTargetInline</a> that is
+		associated with |snapcontainer|.
+	1. Let <var>newBlockTarget</var> be the <a>eventual snap target</a> that |snapcontainer| will snap
+		to along the block axis or null if it would not snap to any element.
+	1. Let <var>newInlineTarget</var> be the <a>eventual snap target</a> that |snapcontainer| will snap
+		to along the inline axis or null if it would not snap to any element.
+	1. If <var>newBlockTarget</var> is not the same element as <var>blockTarget</var>:
+		1. Set the <a>snapchangingTargetBlock</a> associated with |snapcontainer| to
+			<var>newBlockTarget</var>.
+		1. If |snapcontainer| is not already in <var>doc</var>'s
+			<a>pending snapchanging event targets</a>,
+			1. Append |snapcontainer| to <var>doc</var>'s
+				<a>pending snapchanging event targets</a>
+	1. If <var>newInlineTarget</var> is not the same element as <var>inlineTarget</var>:
+		1. Set the <a>snapchangingTargetInline</a> associated with |snapcontainer| to
+			<var>newInlineTarget</var>.
+		1. If |snapcontainer| is not already in <var>doc</var>'s
+			<a>pending snapchanging event targets</a>,
+			1. Append |snapcontainer| to <var>doc</var>'s
+				<a>pending snapchanging event targets</a>.
+
+
+	<h4 id="snap-events-on-layout-changes"> Snap Events due to Layout Changes </h4>
+	When a <a spec=css-scroll-snap lt="scroll snap container">snap container</a>,
+	|snapcontainer|, [[css-scroll-snap-1#re-snap|re-snaps]], run these steps:
+
+	1. Let <var>newBlockTarget</var> be the element that |snapcontainer| has
+		 <a spec="css-scroll-snap-1" lt="scroll snap">snapped</a>  to
+		 in the block axis or null if it did not snap to any element.
+	1. Let <var>newInlineTarget</var> be the element that |snapcontainer| has
+		 <a spec="css-scroll-snap-1" lt="scroll snap">snapped</a> to
+		 in the inline axis or null if it did not snap to any element.
+	1. Let <var>blockTarget</var> be the <a>snapchangingTargetBlock</a> associated
+		with |snapcontainer|.
+	1. Let <var>inlineTarget</var> be the <a>snapchangingTargetInline</a> associated
+		with |snapcontainer|.
+	1. Let <var>snap targets changed</var> be a boolean flag that is initially false.
+	1. If <var>newBlockTarget</var> is not the same element as <var>blockTarget</var>:
+		1. Set the <a>snapchangingTargetBlock</a> associated with |snapcontainer| to
```

Comment:
This looks like it's mostly repeating the snapchanging steps, I think we could invoke the update snapchanging targets algorithm. We can introduce an argument to the algorithm if needed to handle any necessary differences in behavior.

Reply:
Done. I've introduced parameters newBlockTarget and newInlineTarget. For scrolling operation, they are set to the eventual snap targets. For layout changes they are set to the current snap targets.